packages/sui-react-router/src/Router.js

```diff
@@ -82,7 +83,7 @@ const Router = ({
   }, []) // eslint-disable-line
 
   return (
-    <RouterContext.Provider value={state}>
+    <RouterContext.Provider value={{...state, ...(fakeRouter && {router: fakeRouter})}}>
```

Comment:
This could be problematic on production code as we're creating a new object in every render of this provider. In fact, there's already problems with the line 77 because of that.

Reply:
Can we use the `router` prop to essentially do the same that the `fakeRouter` is trying to do?

Reply:
I preferred not to change the router until just passing it to the Provider, because I am not very clear what consequences it could have if there is some kind of navigation in the studio.

To avoid what you are saying, the best thing would be to mutate the state just before assigning it to the provider. That way we wouldn't create a new object every time.